v2/internal/testcommon/vcr/v3/tracking_roundtripper.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v3
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon/vcr"
+)
+
+// Wraps an inner HTTP roundtripper to add a
+// counter for duplicated request URIs. This
+// is then used to match up requests in the recorder
+// - it is needed as we have multiple requests with
+// the same Request URL and it will return the first
+// one that matches.
+type requestCounter struct {
+	inner http.RoundTripper
+
+	countsMutex sync.Mutex
+	counts      map[string]uint32
+}
+
+func AddTrackingHeaders(inner http.RoundTripper) *requestCounter {
+	return &requestCounter{
+		inner:       inner,
+		counts:      make(map[string]uint32),
+		countsMutex: sync.Mutex{},
+	}
+}
+
+const (
+	COUNT_HEADER string = "TEST-REQUEST-ATTEMPT"
```

Comment:
minor: Is the `string` required here? For some reason I thought you could elide the type in this instance.

Reply:
These were **var** and I changed them. Now simplified further.